src/NLog/MappedDiagnosticsLogicalContext.cs

```diff
@@ -129,9 +132,17 @@ public static string Get(string item, IFormatProvider formatProvider)
         /// <returns>The value of <paramref name="item"/>, if defined; otherwise <c>null</c>.</returns>
         public static object GetObject(string item)
         {
-            object value;
-            GetLogicalThreadDictionary().TryGetValue(item, out value);
+            if (!GetLogicalThreadDictionary().TryGetValue(item, out var value))
+                return null;
+
+#if NET4_6 || NETSTANDARD
             return value;
+#else
+            if (value is System.Runtime.Remoting.ObjectHandle objectHandle)
+                return objectHandle.Unwrap();
```

Comment:
why is this not needed for .NET 4.6?

Reply:
Because then I'm using AsyncLocal. Because NET Remoting is deprecated technology.

Reply:
👍

Reply:
But this ofcourse require us to release a NET46 build of NLog to take effect :)

Reply:
owjah yes indeed. 

More build time :'(